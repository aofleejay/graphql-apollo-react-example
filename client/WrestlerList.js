import React from 'react'
import { gql, graphql } from 'react-apollo'
import WrestlerItem from './WrestlerItem'

const WrestlerList = (props) => (
  <section className="section">
    <div className="container">
      <h2 className="subtitle">Wrestler List</h2>
      <hr />
      <div className="content">
        <div>
          {
            props.data.loading ?
              'Loading...'
            :
              props.data.wrestler.map(wrestler => (
                <WrestlerItem
                  key={wrestler.id}
                  name={wrestler.name}
                  twitter={wrestler.twitter}
                  companyName={wrestler.company.name}
                />
              ))
          }
        </div>
      </div>
    </div>
  </section>
)

const wrestlers = gql`
  {
    wrestler {
      name
      twitter
      company {
        name
      }
    }
  }
`

export default graphql(wrestlers)(WrestlerList)
