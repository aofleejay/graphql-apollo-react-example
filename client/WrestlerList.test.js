import React, { Component } from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools'
import { gql } from 'react-apollo'
import { ApolloClient } from 'apollo-client'
import { mockNetworkInterface } from 'apollo-test-utils'
import { graphql } from 'graphql'
import { WrestlerList, wrestlers as query } from './WrestlerList'

describe('<WrestlerList /> UI testing', () => {
  let props

  beforeEach(() => {
    props = {
      data: {
        loading: false,
        error: undefined,
        wrestlers: [
          {
            id: '1',
            name: 'John Cena',
            twitter: '@johncena',
          },
          {
            id: '2',
            name: 'Randy Ortan',
            twitter: '@randyortan',
          },
        ],
      },
    }
  })

  it('snapshot', () => {
    const wrapper = renderer.create(<WrestlerList {...props} />)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('find loading', () => {
    props.data.loading = true
    const wrapper = shallow(<WrestlerList {...props} />)

    expect(wrapper.find('#loading')).toHaveLength(1)
  })

  it('find error', () => {
    props.data.error = 'error'
    const wrapper = shallow(<WrestlerList {...props} />)

    expect(wrapper.find('#error')).toHaveLength(1)
  })

  it('find 2 WrestlerItem', () => {
    const wrapper = shallow(<WrestlerList {...props} />)

    expect(wrapper.find('WrestlerItem')).toHaveLength(2)
  })
})

describe('<WrestlerList /> request testing', () => {
  it('request correctly', () => {
    const result = {
      data: {
        wrestlers: [
          {
            name: 'John Cena',
            twitter: '@johncena',
          },
        ],
      },
    }

    const networkInterface = new mockNetworkInterface({
      request: { query },
      result
    })

    const client = new ApolloClient({
      networkInterface,
      addTypename: false
    })

    return client.query({ query })
      .then(response => {
        expect(response.data).toEqual(result.data);
      })
  })
})
