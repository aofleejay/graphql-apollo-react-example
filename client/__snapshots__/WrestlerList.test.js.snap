// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WrestlerList /> UI testing snapshot 1`] = `
<section
  className="section"
>
  <h2
    className="subtitle"
  >
    Wrestler List
  </h2>
  <hr />
  <div
    className="content"
  >
    <div>
      <div
        className="box"
      >
        <article
          className="media"
        >
          <div
            className="media-left"
          >
            <figure
              className="image is-64x64"
            >
              <img
                alt="Image"
                src="http://bulma.io/images/placeholders/128x128.png"
              />
            </figure>
          </div>
          <div
            className="media-content"
          >
            <div
              className="content"
            >
              <p>
                <strong>
                  John Cena
                </strong>
                 
                <small>
                  @
                  @johncena
                </small>
                <br />
              </p>
            </div>
          </div>
        </article>
      </div>
      <div
        className="box"
      >
        <article
          className="media"
        >
          <div
            className="media-left"
          >
            <figure
              className="image is-64x64"
            >
              <img
                alt="Image"
                src="http://bulma.io/images/placeholders/128x128.png"
              />
            </figure>
          </div>
          <div
            className="media-content"
          >
            <div
              className="content"
            >
              <p>
                <strong>
                  Randy Ortan
                </strong>
                 
                <small>
                  @
                  @randyortan
                </small>
                <br />
              </p>
            </div>
          </div>
        </article>
      </div>
    </div>
  </div>
</section>
`;
